azkaban-common/src/main/java/azkaban/executor/Status.java

```diff
@@ -38,36 +47,10 @@
   }
 
   public static Status fromInteger(final int x) {
-    switch (x) {
-      case 10:
-        return READY;
-      case 20:
-        return PREPARING;
-      case 30:
-        return RUNNING;
-      case 40:
-        return PAUSED;
-      case 50:
-        return SUCCEEDED;
-      case 60:
-        return KILLED;
-      case 70:
-        return FAILED;
-      case 80:
-        return FAILED_FINISHING;
-      case 90:
-        return SKIPPED;
-      case 100:
-        return DISABLED;
-      case 110:
-        return QUEUED;
-      case 120:
-        return FAILED_SUCCEEDED;
-      case 130:
-        return CANCELLED;
-      default:
-        return READY;
+    if (numValMap.containsKey(x)) {
```

Comment:
Yeah we can simply use `Map#getOrDefault` because ImmutableMap doesn't even allow null values.

----

`Map#getOrDefault`:

```java
  @Override
  public final V getOrDefault(@Nullable Object key, @Nullable V defaultValue) {
    V result = get(key);
    return (result != null) ? result : defaultValue;
  }
```